src/goto-harness/memory_snapshot_harness_generator.h

```diff
@@ -48,30 +49,92 @@ class memory_snapshot_harness_generatort : public goto_harness_generatort
   {
   }
 
+  /// 1. load memory table from the snapshot
+  /// 2. add initial section to the initial location
+  /// 3. assign global variables their snapshot values (via the harness
+  ///    function)
+  /// 4. insert call of the initial location (with nondet values) into the
+  ///    harness function
+  /// 5. build symbol for the harness functions
+  /// 6. insert harness function into \p goto_model
+  /// \param goto_model: goto model to be modified
+  /// \param harness_function_name: name of the resulting harness function
   void generate(goto_modelt &goto_model, const irep_idt &harness_function_name)
     override;
 
 protected:
+  /// Collect the memory-snapshot specific cmdline options (one at a time)
+  /// \param option: memory-snapshot | initial-location | havoc-variables
+  /// \param values: list of arguments related to a given option
   void handle_option(
     const std::string &option,
     const std::list<std::string> &values) override;
 
+  /// Check that user options make sense:
+  /// On their own, e.g. location number cannot be specified without a function.
+  /// In relation to the input program, e.g. function name must be known via
+  ///   the symbol table.
   void validate_options() override;
 
+  /// Parse the snapshot JSON file and initialise the symbol table
+  /// \param file: the snapshot JSON file
+  /// \param snapshot: the resulting symbol table build from the snapshot
   void
   get_memory_snapshot(const std::string &file, symbol_tablet &snapshot) const;
 
+  /// Turn this:
```

Comment:
I'd start with a one-sentence summary. Otherwise this likely renders awkwardly.

Reply:
Done.